src/ServiceControlInstaller.Engine/FileSystem/FileUtils.cs

```diff
@@ -126,6 +117,60 @@ internal static void UnzipToSubdirectory(Stream zipStream, string targetPath, st
             }
         }
 
+
+        class ZipFilterEvaluator
+        {
+            readonly string[] folderNameSegments;
+            readonly string targetPath;
+
+            static readonly char[] directorySplitChars = new[] { '/', '\\' };
+
+            public ZipFilterEvaluator(string zipFolderNameToExtract, string targetPath)
+            {
+                folderNameSegments = (zipFolderNameToExtract ?? string.Empty).Split(directorySplitChars, StringSplitOptions.RemoveEmptyEntries);
+                this.targetPath = targetPath;
+            }
+
+            public bool Evaluate(ZipEntry zipEntry, out string resultPath)
```

Comment:
This method is doing two things: 
- applying the filter on the zip entry segment - which dictates if the entry should be extracted or not
- calculating the target path to which the entry should be extracted

Should we break up the method into two submethods? Should we change the naming from `Evalute`? Maybe to `ShouldExtract`?

Reply:
Split and renamed